cmake/MOAB_PullAndMake.cmake

```diff
@@ -0,0 +1,85 @@
+# This Macro sets up the download and build of MOAB using ExternalProject
+# few tweak are done in src/dagmc/CMakeLists.txt and src/PyNE/CMakeists.txt 
+# to make sure that MOAB is built before DAGMC.
+MACRO (moab_pull_make MOAB_Version)
+  MESSAGE(STATUS "MOAB will be downloaded and built")
+  
+  # Ensure STATIC LIBS Build
+  IF(DAGMC_BUILD_STATIC_LIBS)
+    MESSAGE(FATAL_ERROR "PULL_INSTALL_MOAB is ONLY compatible with shared libraries.")
+  endIF()
+    
+  # Ensure we have the right variables to build MOAB
+  moab_autobuild_check_deps()
+   
+    
+  INCLUDE(ExternalProject)
+  MESSAGE("HDF5_ROOT: ${HDF5_ROOT}")
+  SET(MOAB_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/moab")
+  SET(MOAB_ROOT "${CMAKE_BINARY_DIR}/moab")
+  SET(MOAB_INCLUDE_DIRS "${MOAB_INSTALL_PREFIX}/include")
+  SET(MOAB_LIBRARY_DIRS "${MOAB_INSTALL_PREFIX}/lib")
+  MESSAGE("MOAB_LIBRARY_DIRS: ${MOAB_LIBRARY_DIRS}")
+  MESSAGE("CMAKE_SHARED_LIBRARY_SUFFIX: ${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  SET(MOAB_LIBRARIES_SHARED "")
+  ExternalProject_Add(MOAB_ep
+    PREFIX ${MOAB_ROOT}   
+    GIT_REPOSITORY https://bitbucket.org/fathomteam/moab.git
+    GIT_TAG ${moab_version}
+    CMAKE_ARGS
```

Comment:
This is the argument I mentioned on the call today. It looks like they do now set that value, but for ideal optimization it would be set to release. Often libraries include debug symbols in the default build for testing consistency.

https://bitbucket.org/fathomteam/moab/src/master/CMakeLists.txt


For this PR I think we should match the `BUILD_TYPE` to the setting for the DAGMC build.

Reply:
will do that